org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/XMLTextDocumentService.java

```diff
@@ -412,7 +423,13 @@ public void updateCompletionSettings(CompletionSettings newCompletion) {
 	}
 
 	public void updateSymbolSettings(XMLSymbolSettings newSettings) {
-		sharedSettings.symbolSettings.setEnabled(newSettings.isEnabled());
+		XMLSymbolSettings symbolSettings = sharedSettings.symbolSettings;
+		symbolSettings.setEnabled(newSettings.isEnabled());
+		String[] newPatterns = newSettings.getExcluded();
+		if(newPatterns != null) {
+			symbolSettings.setExcludedFiles(XMLExcludedSymbolFile.getExcludedSymbolFiles(newPatterns));

```

Comment:
It should be just symbolSettings#setExcluded which does the XMLExcludedSymbolFile.getExcludedSymbolFiles(newPatterns)